flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/generated/GeneratedClass.java

```diff
@@ -49,10 +49,13 @@ protected GeneratedClass(String className, String code, Object[] references) {
 	@SuppressWarnings("unchecked")
 	public T newInstance(ClassLoader classLoader) {
 		try {
-			return (T) compile(classLoader).getConstructor(Object[].class)
-					// Because Constructor.newInstance(Object... initargs), we need to load
-					// references into a new Object[], otherwise it cannot be compiled.
-					.newInstance(new Object[] {references});
+			if (references.length > 0) {
```

Comment:
It not work if the class don't have non-args constructor and `references.length` is 0.
I think all generated class should invoke `newInstance(new Object[] {references})`.

Reply:
I misunderstood before, thanks for reminding.